### PR TITLE
Fix #3099: Add test suite for DrawableBindingAdapters

### LINKED ISSUE #3099: Add test suite for DrawableBindingAdapters
``DrawableBindingAdapters`` doesn't yet have dedicated tests, and should. This was discovered during #2983. A new test suite should be that calls each of the adapters (via databinding) to ensure each adapter method behaves correctly, including that its background opacity is correctly set.

### PR DESCRIPTION
**Pull request by anandwana001**
## Explanation

Fixes #3099  

Screenshot 2021-05-22 at 12 26 59

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by anandwana001** [comment]
Adding @yashraj-01 @prayutsu @FareesHussain @ARJUPTA as a good PR review exercise while the community bonding period is going on.

File: app/src/main/java/org/oppia/android/app/databinding/DrawableBindingAdapters.java [hunk 1]
```diff
@@ -37,28 +35,4 @@ public static void setTopBackgroundDrawable(@NonNull View view, @ColorInt int co
     GradientDrawable background = (GradientDrawable) view.getBackground();
     background.setColor(0xff000000 | colorRgb);
   }
-
-  /** Used to set a bottom rounded-rect background drawable with a data-bound color. */
-  @BindingAdapter("app:bottomRoundedRectDrawableWithColor")
```
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/databinding/DrawableBindingAdapters.java:42]: removed these three items as we are not using these in the app.

File: app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  }
+
+  @Test
+  fun testDrawableBindingAdapters_setBackgroundColor_hasCorrectBackgroundColor() {
+    val v = View(context)
+    setBackgroundColor(v, resourceColor)
+
+    val color = (v.getBackground() as ColorDrawable).color
+    assertThat(color).isEqualTo(
+      0xff000000.toInt() or resourceColor
```
Comment (FareesHussain) [inline app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt:96]: for this and other test cases.

all tests are to check with a single resourceColor but there isn't a false testcase where we compare the resourceColor with some other value.

**Review by FareesHussain — CHANGES_REQUESTED** [review]
@anandwana001 Left a comment PTAL

File: app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>()
+      .inject(this)
+  }
+
+  @Test
+  fun testDrawableBindingAdapters_setBackgroundColor_hasCorrectBackgroundColor() {
+    val v = View(context)
+    setBackgroundColor(v, resourceColor)
```
Comment (prayutsu) [inline app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt:92]: Maybe we can improve the name of the variable to `view`.

File: app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt (hunk 2 above)
Comment (prayutsu) [inline app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt:96]: Shouldn't we make the assertion to match with the unique color?
E.g- It looks like that the test case will pass in such cases where both correspond to different values.

**Review by prayutsu — CHANGES_REQUESTED** [review]
Left some comments. PTAL @anandwana001

File: app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt (hunk 2 above)
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt:96]: nice catch, will add a false case by passing different resources.

File: app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt (hunk 3 above)
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt:92]: sounds readable

File: app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  @Inject
+  lateinit var context: Context
+
+  private var resourceColor: Int = 0
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    resourceColor = context.resources.getColor(R.color.chapterCardShadow, null)
+  }
```
Comment (Arjupta) [inline app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt:82]: This can be converted into a lazy initialisation

**Review by Arjupta — CHANGES_REQUESTED** [review]
@anandwana001 PTAL

**Review by rt4914 — CHANGES_REQUESTED** [review]
Make sure that all the bazel tests pass in this PR. Mostly they won't as I face similar issue in #3066 where the final conclusion was this: https://github.com/oppia/oppia-android/pull/3066#issuecomment-819879523

**Comment by anandwana001** [comment]
Closing this PR as I am not working on this currently.

**Pull request closed without merging**